corehq/apps/zapier/tests/test_polling.py

```diff
@@ -0,0 +1,138 @@
+import json
+from datetime import datetime
+from unittest.mock import patch
+from django.test import SimpleTestCase, RequestFactory
+from django.contrib.auth.models import User
+
+from corehq import privileges
+from corehq.apps.api import resources
+from corehq.apps.api.resources import v0_4
+from corehq.apps.app_manager.models import Application
+from corehq.apps.api.models import ESXFormInstance
+
+from corehq.apps.zapier.api.v0_5 import ZapierApplicationResource, ZapierXFormInstanceResource
+
+
+class XFormPollingTests(SimpleTestCase):
+    def test_can_poll_with_only_zapier_privilege(self):
+        self.forms = [self._create_form(id='7', domain='test-domain')]
+        self.domain_privileges = [privileges.ZAPIER_INTEGRATION]
+        request = self._create_request('test@test.com', 'test-domain', api_key='121212')
+
+        result = self.resource.dispatch('list', request,
+            domain='test-domain', api_name='v0.5', resource_name='form')
+        rows_found = json.loads(result.content.decode('utf8'))['meta']['total_count']
+        self.assertEqual(rows_found, 1)
+
+    def setUp(self):
+        self.resource = ZapierXFormInstanceResource()
+
+        self.domain_privileges = [privileges.ZAPIER_INTEGRATION]
+        self.forms = []
```

Comment:
nit: These seem unnecessary since they are set in the test. Either that or they could be set here and not in the test.

Reply:
As it currently stands, yes, it is unnecessary and redundant. But there was no existing test suite for this method, and so I left a lot on the table that could be tested. Many of the tests that should be written against this endpoint do not care about the domain privileges, and so for those future tests, it's better to leave that detail here. While I could remove the line from the existing test, I think it's important that the relevant details to the test remain in the test, and not in a helper function.  I think in many cases, YAGNI would apply, but I think it's useful to leave the line here as structure for future tests.